products/logs/frontend/components/VirtualizedLogsList/VirtualizedLogsList.tsx

```diff
@@ -0,0 +1,231 @@
+                                )}
+                            >
+                                {log.parsedBody && prettifyJson
+                                    ? JSON.stringify(log.parsedBody, null, 2)
+                                    : log.cleanBody}
+                            </span>
+                            <div className="flex items-center gap-1 shrink-0">
+                                <LemonButton
+                                    size="xsmall"
+                                    noPadding
+                                    icon={pinned ? <IconPinFilled /> : <IconPin />}
+                                    onClick={(e) => {
+                                        e.stopPropagation()
+                                        togglePinLog(log.uuid)
+                                    }}
+                                    tooltip={pinned ? 'Unpin log' : 'Pin log'}
+                                    className={cn(
+                                        pinned ? 'text-warning' : 'text-muted opacity-0 group-hover:opacity-100'
+                                    )}
+                                />
+                                <div className="opacity-0 group-hover:opacity-100" onClick={(e) => e.stopPropagation()}>
+                                    <LogsTableRowActions log={log} />
+                                </div>
+                            </div>
+                        </div>
+                    </div>
+                )}
+            </CellMeasurer>
+        )
+    }
+
+    const handleRowsRendered = ({ stopIndex }: { stopIndex: number }): void => {
+        if (disableInfiniteScroll) {
+            return
+        }
+        if (shouldLoadMore(stopIndex, dataSource.length, hasMoreLogsToLoad, logsLoading)) {
+            fetchNextLogsPage(250)
+        }
+    }
+
+    if (dataSource.length === 0 && !loading) {
+        return <div className="p-4 text-muted text-center">No logs to display</div>
+    }
+
+    // Fixed height mode for pinned logs
+    if (fixedHeight !== undefined) {
+        return (
+            <div style={{ height: fixedHeight }}>
+                <AutoSizer disableHeight>
+                    {({ width }) => (
+                        <List
+                            ref={listRef}
+                            width={width}
+                            height={fixedHeight}
+                            rowCount={dataSource.length}
+                            rowHeight={cache.rowHeight}
+                            deferredMeasurementCache={cache}
+                            rowRenderer={rowRenderer}
+                            overscanRowCount={5}
+                            tabIndex={null}
+                            style={{ outline: 'none' }}
+                        />
+                    )}
+                </AutoSizer>
+            </div>
+        )
+    }
+
+    return (
+        <div className="h-full">
```

Comment:
```suggestion
        <div className="h-screen">
```

make it so we can scroll to view a full screen of logs